python/ember/input.py

```diff
@@ -1314,6 +1342,183 @@ def run(self):
+                        self.terminationCondition.tolerance /= 2.0
+
+                        done = False
+                        while not done:
+                            done = solver.step()
+                            if max(solver.T) < self.extinction.cutoffTemp:
+                                done = True
+                                extinguished = True
+                        solver.finalize()
+                        t2 = time.time()
+
+                    if max(solver.T) >= maxTemps[-1]:
+                        print('Refining tolerance failed')
+                        print('marking as extinguished..')
+                        extinguished = True
+
+            if extinguished is False:
+                log('Found burning solution with Tmax = %.1f' % max(solver.T))
+
+                restartFile = 'prof_sr{:08.2f}.{}'.format(strainRate, fileExt)
+                restartPath = os.path.join(ssProfilePath, restartFile)
+
+                solver.writeStateFile(os.path.splitext(restartFile)[0])
+                os.rename(os.path.join(self.paths.outputDir, restartFile), restartPath)
+
+                strainRateValues.append(strainRate)
+                maxTemps.append(max(solver.T))
+
+                confString = self.original.stringify()
+                configFile = 'conf_extinction.txt'
+                configPath = os.path.join(outDirTop, configFile)
+                with open(configPath, 'w') as confOut:
+                    confOut.write(confString)
+
+            else:
+                log('Found non-burning solution')
+
+                if restartPath is not None:
+
+                    hasExtinguished = True
+
+                    if self.extinction.method is 'step':
+                        strainRate -= stepSize
+                        stepSize *= self.extinction.reductionFactor
+                        if stepSize < self.extinction.minStep:
+                            complete = True
+                    else:
+                        strainRate /= incFactor
+                        incFactor = 1.0 + (incFactor-1.0) * self.extinction.reductionFactor
+                        if incFactor < self.extinction.minFactor:
+                            complete = True
+
+            if complete is False:
+                shutil.rmtree(currentRunPath)
+                os.mkdir(currentRunPath, 0o0755)
+
+            if restartPath is None:
+                strainRate /= 2.0
+                if strainRate < 10.0:
+                    complete = True
+                    print('Failed to find burning starting point')
+            else:
+                self.readInitialCondition(restartPath)
+
+        with open(os.path.join(outDirTop,'extProfile.csv'),'w') as extProgData:
+            extProgData.write('Strain Rate [1/s],Max. Temp. [K],\n')
+
+            for i in range(len(strainRateValues)):
+                extProgData.write('{0:0.5f},{1:0.5f},\n'.format(strainRateValues[i],maxTemps[i]))
```

Comment:
What's the purpose of the trailing comma? I thought that was generally not desirable in CSV files.

Reply:
No purpose, this is just a mistake.